docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,228 @@ The remaining points can happen any time after 1) is complete although note that
 
-5. Handle `rose-suite.conf` functionality
-   * Pull items from other locations / sources on installation (e.g. svn).
-   * Configure Jinja2 / EmPy inputs.
-   * Set environment variables exported to the suite daemon?
-   * Continue to handle `rose-suite.conf` using a plugin preprocessor.
-   * Decide which features to support natively in Cylc and how.
+5. Handle
+   [rose-suite.conf](https://metomi.github.io/rose/doc/html/api/configuration/suite.html#rose:file:rose-suite.conf)
+   functionality including optional configurations. See
+   [cylc-flow #3819](https://github.com/cylc/cylc-flow/issues/3819)
+   and [rose #2412](https://github.com/metomi/rose/pull/2412).
 
 6. Utility to remove installed locations (on all platforms) on demand
    (equivalent to `rose suite-clean`).
 
-7. On starting up a new run of the suite assign a new run name:
+7. Use a different run directory for each installation of a suite:
    ```
-   mi-aa001/
+   u-ab123/
      run1
      run2
    ```
-   * This is new functionality but Cylc 8 is a good time to introduce it.
+   * This is big change in working practise but Cylc 8 is a good time to
+     introduce it.
    * Use a symbolic link to identify the latest run which is the default name
      when performing operations on a suite?
 
-8. Consider the behaviour created by the `CYLC_VERSION` and `ROSE_VERSION`
-   variables in `rose-suite.conf`. These allow rose suite-run to:
-   * Change the version of Cylc and check the version of Rose used to run the
-     suite.
-   * Prevent suites accidentally upgrading or downgrading versions on restart.
-
-9. Consider `rose prune` functionality.
+8. Consider `rose prune` functionality.
    * Need to make sure it will continue to work correctly.
-   * Need Cylc equivalent but this should be done as part of
+   * Longer term we should build housekeeping into Cylc, see
      [cylc-flow #1159](https://github.com/cylc/cylc-flow/issues/1159).
 
-### CLI
+9. Record version control information when you install a suite, see
+   [cylc-flow #3849](https://github.com/cylc/cylc-flow/issues/3849).
+   * `rose suite-run` current supports this for Subversion only
+     (`svn info` + `svn status` + `svn diff`).
+   * Extend this to support equivalent information for git.
+
+### CLI changes
+
+`cylc install` - new command to install workflows.
+* By default, `cylc install` will install the workflow found in `$PWD` into
+  `~/cylc-run/$(basename $PWD)/runN` (where `runN` = `run1`, `run2` ...).
+* If `run1` already exists it will install into `run2` (and so on).
+  Create a symlink `runN` pointing at the latest run.
+* `--run-name=my-run` implies install into `~/cylc-run/$(basename $PWD)/my-run`.
+  If the target directory already exists then fail.
+* `--no-run-name` implies install into `~/cylc-run/$(basename $PWD)`.
+* `--flow-name=my-flow` implies install into `~/cylc-run/my-flow/runN`.
+* `--directory=/path/to/flow` (`-C ...`) implies install the workflow found in
```

Comment:
consider aliases `--source-dir`, `--src-dir`?

Reply:
I think this is ok.

This option effectively specifies the path to a Cylc configuration file/dir so follows the pattern of commands like `make` which uses `--directory`, `-C`.